File: 3. Arrays/1. Easy/C++/5. Left rotate array by one.cpp
/*
Left rotate elements of array by 1 place
*/

#include <bits/stdc++.h>
using namespace std;


void left_rotate_by_one(int arr[],int n){
    // code here
    int firstElement = arr[0];
    for(int i=0;i<n-1;i++){
        arr[i] = arr[i+1];
    }
    arr[n-1] = firstElement;
}

int main() {
	// your code goes here
    int n;
    cin >> n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin >> arr[i];
    }
    left_rotate_by_one(arr,n);
    for(int i=0;i<n;i++){
        cout << arr[i] << " ";
    }
}

File: 3. Arrays/1. Easy/Python/5. Left rotate array by one.py
# cook your dish here
#Left rotate elements of array by 1 place
import sys

def leftRotateByOne(arr,n):
    firstElement = arr[0]
    for i in range(0,n-1):
       arr[i] = arr[i+1]
    arr[n-1] = firstElement


n = int(input())
arr = list(map(int,input().strip().split()))
leftRotateByOne(arr,n)
for i in range(0,n):
    print(arr[i],end=" ")